probe-rs/src/probe/debug_probe.rs

```diff
@@ -9,8 +14,14 @@ use log::debug;
 use crate::memory::adi_v5_memory_interface::ADIMemoryInterface;
 use crate::memory::MI;
 use crate::probe::protocol::WireProtocol;
+use colored::*;
 use std::error::Error;
 use std::fmt;
+use std::time::Instant;
+
+const UNLOCK_TRIES: usize = 2;
+const UNLOCK_TIMEOUT: u64 = 15;
+const CTRL_AP_IDR: u32 = 0x0288_0000;
```

Comment:
This should really be an `IDR` struct for readability, imo and not just a magic value.